lib/eblox/data/taxonomy.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Eblox.Data.Taxonomy do
+  @moduledoc false
+
+  use Supervisor
+
+  @type post_id() :: term()
+  @type registry_option() :: Registry.start_option()
+
+  @callback registry_options([registry_option()]) :: [registry_option()]
+  @callback on_add(module(), post_id(), term()) :: :ok | :error
+  @callback on_remove(module(), post_id()) :: :ok
+
+  defmodule Meta do
+    @moduledoc false
+
+    use Agent
+
+    alias Eblox.Data.Taxonomy.Meta
+
+    @type t :: %{
+      __struct__: Meta,
+      impl: module(),
+    }
+    @enforce_keys ~w|impl|a
+    defstruct ~w|impl|a
+
+    def start_link([{:impl, impl} | opts]) do
+      Agent.start_link(fn -> %Meta{impl: impl} end, opts)
+    end
+
+    def impl(name) do
+      Agent.get(name, &(&1 |> Map.get(:impl)))
```

Comment:
There is an unwritten rule, pushed by José: pipe is _a pipe_, and it should be used when there is a _pipeline_. A single call should not be piped (`Map.get(&1, :impl)`.) Also, you have a struct on hand, why would you need `Map.get/2` at all?
```suggestion
      Agent.get(name, & &1.impl)
```

Reply:
I instinctively followed this rule by changing a few pipes to regular function calls but I must have missed this one! :–)